index.js

```diff
@@ -1,35 +1,35 @@
 'use strict';
+const PassThrough = require('stream').PassThrough;
 const zlib = require('zlib');
 
 module.exports = res => {
 	// TODO: use Array#includes when targeting Node.js 6
 	if (['gzip', 'deflate'].indexOf(res.headers['content-encoding']) !== -1) {
 		const unzip = zlib.createUnzip();
+		const stream = new PassThrough();
 
-		unzip.httpVersion = res.httpVersion;
-		unzip.headers = res.headers;
-		unzip.rawHeaders = res.rawHeaders;
-		unzip.trailers = res.trailers;
-		unzip.rawTrailers = res.rawTrailers;
-		unzip.setTimeout = res.setTimeout.bind(res);
-		unzip.statusCode = res.statusCode;
-		unzip.statusMessage = res.statusMessage;
-		unzip.socket = res.socket;
+		stream.httpVersion = res.httpVersion;
+		stream.headers = res.headers;
+		stream.rawHeaders = res.rawHeaders;
+		stream.trailers = res.trailers;
+		stream.rawTrailers = res.rawTrailers;
+		stream.setTimeout = res.setTimeout.bind(res);
+		stream.statusCode = res.statusCode;
+		stream.statusMessage = res.statusMessage;
+		stream.socket = res.socket;
 
-		unzip.once('error', err => {
+		unzip.on('error', err => {
 			if (err.code === 'Z_BUF_ERROR') {
-				res.emit('end');
+				stream.end();
 				return;
 			}
 
-			res.emit('error', err);
+			stream.emit('error', err);
 		});
 
-		res.on('close', () => {
-			unzip.emit('close');
-		});
+		res.pipe(unzip).pipe(stream);
 
-		res = res.pipe(unzip);
+		res = stream;
```

Comment:
Would be better if we just returned `stream` below instead of overwriting `res`.